python/ray/ml/checkpoint.py

```diff
@@ -332,12 +330,36 @@ def to_directory(self, path: Optional[str] = None, dedup: bool = True) -> str:
                     shutil.copytree(local_path, path)
             elif external_path:
                 # If this exists on external storage (e.g. cloud), download
-                download_from_uri(uri=external_path, local_path=path)
+                download_from_uri(uri=external_path, local_path=path, filelock=False)
             else:
                 raise RuntimeError(
                     f"No valid location found for checkpoint {self}: {self._uri}"
                 )
 
+    def to_directory(self, path: Optional[str] = None) -> str:
+        """Write checkpoint data to directory.
+
+        Args:
+            path: Target directory to restore data in. If not specified,
+                will create a temporary directory.
+
+        Returns:
+            str: Directory containing checkpoint data.
```

Comment:
Document deduping behavior (should we make it an option)?

Reply:
I think it is fine to keep it as is. It's a special case only for ObjectRef. I modified `to_directory` docstring but I can mention it here too